homeassistant/components/speedtestdotnet/__init__.py

```diff
@@ -31,13 +31,14 @@ async def _request_refresh(event: Event) -> None:
         """Request a refresh."""
         await coordinator.async_request_refresh()
 
-    if hass.state == CoreState.running:
-        await coordinator.async_config_entry_first_refresh()
-    else:
-        # Running a speed test during startup can prevent
-        # integrations from being able to setup because it
-        # can saturate the network interface.
-        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STARTED, _request_refresh)
+    if not config_entry.pref_disable_polling:
```

Comment:
This should not be handled here IMHO, we already handling polling preferences in the coordinator, to me it seems like this is a miss.

Reply:
Here we are disabling the initial refresh at startup. To my understanding even if polling is disabled in system options the initial coordinator refresh is executed for each integration that uses the coordinator at startup. For this integration we were adding an option to skip the initial refresh if polling is disabled (leaving the user with automating when the refresh should happen).